js/data.js

```diff
@@ -6,3 +6,9 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+window.addEventListener('beforeunload', logBefore);
+
+function logBefore(event) {
+  window.localStorage.setItem('javascript-local-storage', JSON.stringify(data));
```

Comment:
You can just say `localStorage`, it's unnecessary to say `window.localStorage`. Also, what's the name of that property you're saving in localStorage? That doesn't seem descriptive of this data 🤔